notebooks/contrib-dev/Stochastic_Simu_Chem_Rxn.ipynb

```diff
@@ -0,0 +1,2654 @@
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## 3b-1.1 - Forward Euler Method"
+      ],
+      "metadata": {
+        "id": "IZGXRXGFaUL-"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The Forward Euler Method is a numerical technique for approximating the solutions to ordinary differential equations (ODEs). It is well known to be a simple and straightforward method that's useful for solving ODEs when you can't find an exact analytical solution.\n",
+        "\n",
+        "Basically, it approximates the solution to a first-order ordinary differential equation of the form:\n",
+        "\n",
+        "\\begin{equation}\n",
+        "\\frac{dy}{dx} = f(x, y)\n",
+        "\\end{equation}\n",
+        "\n",
+        "The Forward Euler Method also works through an iterative approach that breaks down the continuous ODE into discrete steps, by using the following formula to update the solution at each step:\n",
+        "\n",
+        "\\begin{equation}\n",
+        "y_(n+1) = y_n + h * f(x_n, y_n)\n",
+        "\\end{equation}\n",
+        "\n",
+        "In this formula,\n",
+        "\n",
+        "$y_(n+1)$ is the new approximation of the solution at the next step.\n",
+        "\n",
+        "\n",
+        "$y_n$ is the previous approximation of the solution at the current step.\n",
+        "\n",
+        "$h$ is the step size, which represents the size of the interval over which you're approximating the solution.\n",
+        "\n",
+        "$x_n$ is the current value of the independent variable at the nth step.\n",
+        "\n",
+        "$f(x_n, y_n)$ is the rate of change of y at the current point $(x_n, y_n).$\n",
+        "\n",
+        "This way, by applying the formula and"
+      ],
+      "metadata": {
+        "id": "rbgjRVf7aVcq"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Now, implement the found differential equations into the Forward Euler Method to find and plot the  numerically solution.**"
+      ],
+      "metadata": {
+        "id": "bQllXd0omV3H"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import numpy as np\n",
+        "import matplotlib.pyplot as plt\n",
+        "\n",
+        "# Define the parameters and initial conditions\n",
+        "k1 = 0.5\n",
+        "k2 = 1.0\n",
+        "xa0 = 1000.0  # Initial concentration of Ca\n",
+        "t_max = 16.0  # Maximum time\n",
+        "h = 0.1  # Time step\n",
+        "\n",
+        "# Create arrays to store the results\n",
+        "t_values = np.arange(0, t_max + h, h)\n",
+        "Ca_values = [xa0]  # Initial value\n",
+        "Cb_values = [0.0]  # Initial value\n",
+        "Cc_values = [0.0]  # Initial value\n",
+        "\n",
+        "# Numerically solve the differential equations using Forward Euler Method\n",
+        "\n",
+        "### BEGIN SOLUTION ###\n",
+        "for t in t_values[:-1]:\n",
```

Comment:
This should instead be the generic method implemented as a function. Then, you apply the function to this specific problem. This separates the logic of a generic method from the specific problem. It is also the best practice we used in the class (see other notebooks).